lib/modules/orangered.js

```diff
@@ -205,48 +271,112 @@ const setupFaviconBadge = _.once(() => {
 const orangeredElements = () => ({
 	...nativeButtons(),
 	...floatingButtons(),
-	modmailButtonSelector: '#modmail',
 });
 
 const $orangeredElements = () => Object.values(orangeredElements()).reduce(($all, ele) => $all.add(ele), $());
 
-const nativeButtons = _.once(() => {
-	const nativeInboxButton = document.querySelector('#header-bottom-right #mail');
-	let nativeInboxCount = document.querySelector('#header-bottom-right .message-count');
-	if (!nativeInboxCount && (module.options.updateCurrentTab.value || module.options.updateOtherTabs.value)) {
+const nativeButtons = _.once(() => _nativeButtons(document.body, true));
+const _nativeButtons = (container = document.body, addIfMissing) => {
+	const nativeMessageButton = container.querySelector('#header-bottom-right #mail, #header-bottom-right #message');
+	const nativeMessageCount = getOrAddInboxCount(container, nativeMessageButton, 'message-count', addIfMissing);
+	const nativeNotificationButton = container.querySelector('#header-bottom-right #notification');
+	const nativeNotificationCount = getOrAddInboxCount(container, nativeNotificationButton, 'inbox-count', addIfMissing);
+	const nativeModmailButton = container.querySelector('#modmail');
+	const nativeNewModmailButton = container.querySelector('#new_modmail');
+
+	return {
+		nativeMessageButton,
+		nativeMessageCount,
+		nativeNotificationButton,
+		nativeNotificationCount,
+		nativeModmailButton,
+		nativeNewModmailButton,
+	};
+};
+
+function getOrAddInboxCount(container, nativeInboxButton, countClassname, addIfMissing) {
+	let nativeInboxCount = container.querySelector(`#header-bottom-right .${countClassname}`);
+	if (addIfMissing && nativeInboxButton && !nativeInboxCount && (module.options.updateCurrentTab.value || module.options.updateOtherTabs.value)) {
 		nativeInboxCount = document.createElement('a');
+		nativeInboxCount.classList.add(countClassname);
 		nativeInboxCount.style.display = 'none';
 		$(nativeInboxButton).after(nativeInboxCount);
 	}
-	return { nativeInboxButton, nativeInboxCount };
-});
 
-const floatingButtons = _.once(() => {
-	if (!module.options.showFloatingEnvelope.value) return {};
-	let $floatingInboxButton, $floatingInboxCount;
+	return nativeInboxCount;
+}
 
+const floatingButtons = _.once(() => {
+	const DEFAULT = {};
+	if (!module.options.showFloatingEnvelope.value) return DEFAULT;
 	const pinHeader = BetteReddit.module.options.pinHeader.value;
-	if (pinHeader === 'sub' || pinHeader === 'none') {
-		$floatingInboxButton = $('<a>', { id: 'NREMail' });
-		Floater.addElement($floatingInboxButton);
-
-		$floatingInboxCount = $('<a>', { id: 'NREMailCount' });
-		$floatingInboxCount.css('display', 'none');
-		$floatingInboxCount.attr('href', getInboxLink(true));
-		Floater.addElement($floatingInboxCount);
+	if (!(pinHeader === 'sub' || pinHeader === 'none')) {
+		return DEFAULT;
 	}
 
-	return { $floatingInboxButton, $floatingInboxCount };
+	const nb = nativeButtons();
+
+	const nativeButtonList = [
```

Comment:
I explicitly listed them to control the ordering because I don't trust `Object.keys()` to be consistent cross-browser.